### PR TITLE
added fetchAnnouncements & reworked func structure

### PR DESCRIPTION
**Pull request by Take434**

### CODE REVIEW

**Review by rdmchr — CHANGES_REQUESTED** [review]
Missing comments for most functions

File: apps/functions/src/fetchAnnouncements.ts [hunk 1]
```diff
@@ -36,38 +36,48 @@ export const fetchAnnouncements =
...
+    /* eslint-disable */
      await fetchData(userData.xUser, userData.xPass, uid, URL, sel, endpoint, action);
     return parseJSON(JSON.stringify(announcements), uid);
-});
-
+  });
 
+/**
+ * 
+ * @param messages this is messages
```
Comment (rdmchr) [inline apps/functions/src/fetchAnnouncements.ts:47]: ?

File: apps/functions/src/fetchAnnouncements.ts [hunk 2]
```diff
@@ -36,38 +36,48 @@ export const fetchAnnouncements =
...
-        await store.collection("messages").doc(""+messageId).update({
-            permittedUsers : FieldValue.arrayUnion(uid)
-            });
-        console.log('have tried');
+    if (coll.exists) {
+      console.log("still trying");
+      await store.collection("messages").doc(""+messageId).update({
+        permittedUsers: FieldValue.arrayUnion(uid),
+      });
+      console.log("have tried");
```
Comment (rdmchr) [inline apps/functions/src/fetchAnnouncements.ts:75]: Remove useless `console.log()` statements

**Review by rdmchr — CHANGES_REQUESTED** [review]

File: apps/functions/src/fetchAnnouncements.ts (hunk 1 above)
Comment (Take434) [reply apps/functions/src/fetchAnnouncements.ts:47]: ?

File: apps/functions/src/fetchTimetable.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    console.log(day);
+    week[dayNum] = day;
+  }
+  uploadToFirestore(week, uid);
+  return week;
+}
+
+/**
+ * 
+ * @param data this is data
```
Comment (rdmchr) [inline apps/functions/src/fetchTimetable.ts:83]: Still missing some comments

**Review by rdmchr — APPROVED** [review]

**Pull request merged**